Lib/fontmake/__main__.py

```diff
@@ -161,12 +161,23 @@ def main(args=None):
              '%(choices)s. Default: INFO')
     args = vars(parser.parse_args(args))
 
-    project = FontProject(timing=args.pop('timing'),
-                          verbose=args.pop('verbose'))
-
     glyphs_path = args.pop('glyphs_path')
     ufo_paths = args.pop('ufo_paths')
     designspace_path = args.pop('mm_designspace')
+
+    if 'variable' in args['output']:
+        if not (glyphs_path or designspace_path):
+            parser.error(
+                'Glyphs or designspace source required for variable font')
```

Comment:
I'm not sure but why can't we generate VF from ufo file?
IIUC, glyphs are converted to ufo file, then ufo file is compiled to binary.